src/srcset.js

```diff
@@ -197,10 +197,10 @@ export class Srcset {
    * @private
    */
   selectByWidth_(width, dpr) {
-    let minIndex = -1;
-    let minWidth = 1000000;
-    let minScore = 1000000;
-    for (let i = 0; i < this.sources_.length; i++) {
+    const length = this.sources_.length;
+    let minWidth = this.sources_[length - 1].width / dpr;
```

Comment:
Technically, if you specify `w` for `srcset` - they are required for all srcset. In other words, having one absent is an error. However, these cases are seen commonly in practice and in fact all of the browsers tolerate them. Since we have no idea what width it is, the only assumption that it's smaller than the rest and we assume it's half the minimum width.